crates/ty_server/src/server/main_loop.rs

```diff
@@ -113,8 +141,8 @@ impl Server {
     /// Returns `Ok(None)` if the client connection is closed.
     fn next_event(&self) -> Result<Option<Event>, crossbeam::channel::RecvError> {
         let next = select!(
-            recv(self.connection.incoming()) -> msg => msg.map(Event::Message),
-            recv(self.main_loop_receiver) -> event => return Ok(event.ok()),
+            recv(self.connection.receiver) -> msg => return Ok(msg.ok().map(Event::Message)),
```

Comment:
Why are we ignoring the error when receiving the client messages (via `connection`) but not for the main loop receiver?

Reply:
Oh, is it to explicit that the connection got closed without a proper shutdown sequence which is how it's being handled in the main loop?

Reply:
Yes, we ignore the connection closed errors for the IO channel because that means the server exited in a non-graceful way. 

We should never see those for the main loop because we always have at least our own sender.